diff --git a/native/src/core/module.cpp b/native/src/core/module.cpp
index 94dea6c8d..c813c605e 100644
--- a/native/src/core/module.cpp
+++ b/native/src/core/module.cpp
@@ -283,7 +283,7 @@ void load_modules() {
 
     if (!system->is_empty()) {
         // Handle special read-only partitions
-        for (const char *part : { "/vendor", "/product", "/system_ext" }) {
+        for (const char *part : { "/vendor", "/product", "/system_ext", "/odm", "/my_bigball", "/my_carrier", "/my_company", "/my_engineering", "/my_heytap", "/my_manifest", "/my_preload", "/my_product", "/my_region", "/my_stock", "/vendor_dlkm" }) {
             struct stat st{};
             if (lstat(part, &st) == 0 && S_ISDIR(st.st_mode)) {
                 if (auto old = system->extract(part + 1)) {
